cmd/lotus-storage-miner/info.go

```diff
@@ -157,16 +164,46 @@ func infoCmdAct(cctx *cli.Context) error {
 	if !pow.HasMinPower {
 		fmt.Print("Below minimum power threshold, no blocks will be won")
 	} else {
-		expWinChance := float64(types.BigMul(qpercI, types.NewInt(build.BlocksPerEpoch)).Int64()) / 1000000
-		if expWinChance > 0 {
-			if expWinChance > 1 {
-				expWinChance = 1
-			}
-			winRate := time.Duration(float64(time.Second*time.Duration(build.BlockDelaySecs)) / expWinChance)
-			winPerDay := float64(time.Hour*24) / float64(winRate)
 
-			fmt.Print("Expected block win rate: ")
-			color.Blue("%.4f/day (every %s)", winPerDay, winRate.Truncate(time.Second))
+		winRatio := new(corebig.Rat).Mul(
+			new(corebig.Rat).SetFrac(
+				types.BigMul(pow.MinerPower.QualityAdjPower, types.NewInt(build.BlocksPerEpoch)).Int,
+				pow.TotalPower.QualityAdjPower.Int,
+			),
+			// decrease the rate ever-so-slightly to very roughly account for the multi-win poisson distribution
+			// FIXME - this is not a scientifically derived number... like at all
+			corebig.NewRat(99997, 100000),
+		)
+
+		if winRatioFloat, _ := winRatio.Float64(); winRatioFloat > 0 {
+
+			weekly, _ := new(corebig.Rat).Mul(
+				winRatio,
+				new(corebig.Rat).SetInt64(7*builtin.EpochsInDay),
+			).Float64()
+
+			avgDuration, _ := new(corebig.Rat).Mul(
+				corebig.NewRat(int64(time.Second)*builtin.EpochDurationSeconds, 1),
+				new(corebig.Rat).Inv(winRatio),
+			).Float64()
+
+			fmt.Print("Projected average block win rate: ")
+			color.Blue(
+				"%.02f/week (every %s)",
+				weekly,
+				time.Duration(avgDuration).Truncate(time.Second).String(),
+			)
+
+			// Geometric distribution calculated as described in https://en.wikipedia.org/wiki/Geometric_distribution#Probability_Outcomes_Examples
+			// https://www.wolframalpha.com/input/?i=c%3D99%3B+p%3D188809111007232%3B+n%3D5740343177447735296%3B+%281-%284.99*p%2Fn%29%29%5E%28t*2880%29%3D%281-%28c%2F100%29%29
+			fmt.Print("Projected block win with ")
+			color.Green(
+				"99.9%% probability every %s",
+				time.Duration(int64(
+					float64(time.Second*builtin.EpochDurationSeconds)*math.Log(1-0.999)/math.Log(1-winRatioFloat),
+				)).Truncate(time.Second).String(),
+			)
```

Comment:
@Kubuxu I hope I got this correctly...

Reply:
@Kubuxu note - the above is an outdated version of the code it was slightly reworked since ( but the main log() I am not sure about is the same )

Reply:
@Kubuxu more specifically: 
I do:
`( 1 - winrate ) ^ time = 0.001`

But the wiki page seems to indicate I should
`( 1 - winrate ) ^ time = 0.001 / winrate`

... which gives me a negative number :|